components/renku_data_services/storage/models.py

```diff
@@ -222,3 +222,28 @@ def _from_ambiguous_url(
 
         # default to S3 for unknown URLs, since these are way more common
         return CloudStorage.from_s3_url(storage_url, project_id, name, readonly, target_path)
+
+
+class CloudStorageSecret(BaseModel):
+    """Cloud storage secret model."""
+
+    user_id: str = Field()
+    storage_id: str = Field()
+    name: str = Field(min_length=1, max_length=99)
+    secret_id: str = Field()
+
+    @classmethod
+    def from_dict(cls, data: dict) -> "CloudStorageSecret":
+        """Create the model from a plain dictionary."""
+        if "user_id" not in data:
+            raise errors.ValidationError(message="'user_id' not set")
+        if "storage_id" not in data:
+            raise errors.ValidationError(message="'storage_id' not set")
+        if "name" not in data:
+            raise errors.ValidationError(message="'name' not set")
+        if "secret_id" not in data:
+            raise errors.ValidationError(message="'secret_id' not set")
```

Comment:
question: doesn't pydantic already validate this? also I don't see this method used anywhere in this PR.

Reply:
Most of these classes had validation and I'd also did the same here. Removed.